src/Microsoft.Tye.Hosting/TyeHost.cs

```diff
@@ -59,6 +60,15 @@ public async Task RunAsync()
             await StopAsync();
         }
 
+        public Task PurgeAsync()
+        {
+            var app = BuildWebApplication(_application, _args, Sink);
+            var runners = CreateRunners(_application, _args, app.Logger, app.Configuration);
+
+            using var replicaRegistry = new ReplicaRegistry(_application, runners);
+            return replicaRegistry.Reset();
```

Comment:
meta-question:

All this work is done to preserve the layering that each kind of runner knows how to clean up it's own services, and that is expressed in the abstraction now.

As a result, we've got to read all of the configuration, and create the runners, and create a registry just so we can loop through the events and kill processes. 

I don't think it's terribly wrong, it's just different from what I expected when I think about "cleanup" of lingering services.